### ISSUE TITLE
[Bug Report] Source data is not displayed when using log crate

### ISSUE THREAD

**Issue by alex-lavriv** [SpriteOvO/spdlog-rs#22]
## Describe the bug
 Source data is not displayed when using log crate, but displayed properly when using spdlog directly.

## To Reproduce

**main.rs**
```
use spdlog::Logger;
use spdlog::prelude::*;

use std::sync::Arc;
fn main() {
    use_pattern_formatter();
    spdlog::init_log_crate_proxy()
        .expect("users should only call `init_log_crate_proxy` function once");

    let custom_logger: Arc<Logger> = spdlog::default_logger();
    let proxy: &'static spdlog::LogCrateProxy = spdlog::log_crate_proxy();
    proxy.set_logger(Some(custom_logger));
    log::set_max_level(log::LevelFilter::Trace);
    log::info!("No source data");
    info!("With source data");
}

fn use_pattern_formatter() {
    use spdlog::{
        formatter::{pattern, PatternFormatter},
    };
    let new_formatter: Box<PatternFormatter<_>> = Box::new(PatternFormatter::new(pattern!(
        "{date}T{time}.{millisecond}Z [{^{level}}] ({module_path}::{file_name}) {payload}{eol}"
    )));

    for sink in spdlog::default_logger().sinks() {
        sink.set_formatter(new_formatter.clone())
    }
}

```
**Cargo.toml**
```
[package]
name = "log_test"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
log = "0.4.17"
spdlog-rs = {version = "0.3.7", features = ["source-location", "log"]}
```

**Output**
```
2023-02-05T19:40:42.010Z [info] (::) No source data
2023-02-05T19:40:42.010Z [info] (log_test::main.rs) With source data
```

## Environments

 - OS: Ubuntu 
 - Version: 18.04

**Comment by SpriteOvO** [comment]
Thanks for reporting!

I opened a PR #23 to fix it, could you check if it works for you?

**Comment by alex-lavriv** [comment]
Thanks for the fast response!
Seems to work even without adding `source-location` dependency

**Comment by SpriteOvO** [comment]
> Seems to work even without adding `source-location` dependency

Yes, it makes sense to me. I think the core purpose of the feature `source-location` is to avoid source information appearing in the final binary, however we can't control whether the `log` crate stores the information or not. So if you just don't want to display it, you should remove the related patterns from your formatter. Let me know if you have different thoughts at this point :)

**Comment by alex-lavriv** [comment]
That's fine by me! 
Thank you for the fix! Waiting for the new tag.

**Comment by SpriteOvO** [comment]
Released within [v0.3.8](https://github.com/SpriteOvO/spdlog-rs/releases/tag/v0.3.8).
